WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -182,6 +182,7 @@ final class EditableOrderViewModel: ObservableObject {
             stores: stores,
             supportsMultipleSelection: isProductMultiSelectionEnabled,
             toggleAllVariationsOnSelection: false,
+            topProductsProvider: TopProductsFromCachedOrdersProvider(storageManager: ServiceLocator.storageManager),
```

Comment:
Should we move back the responsibility of providing a default `ServiceLocator.storageManager` to the `TopProductsFromCachedOrdersProvider` class,  if one is not injected directly as a dependency into its initializer? So in `TopProductsFromCachedOrdersProvider`:

```
    init(storageManager: StorageManagerType = ServiceLocator.storageManager) {
        self.storageManager = storageManager
    }
```

And once done, I'm thinking perhaps could be even better to decouple it entirely from the Service Locator by passing the protocol to the `ProductSelectorViewModel`:

```
private let topProductsProvider: ProductSelectorTopProductsProviderProtocol = TopProductsFromCachedOrdersProvider()
```

```suggestion
            topProductsProvider: topProductsProvider,
```

Reply:
Yeah, thanks for that, done in 3078e76c158a73d9df5851bc864e2e5fe11d5392